### PR TITLE
Add integration spec and fix n+1 problems

### PR DESCRIPTION
**Pull request by abiola-arobieke**
In this milestone:

The N+1 problem was solved while fetching all posts and their comments.
Used Capybara to write integration tests for each view in the project.
Attached below is the screenshot before and after solving N+1 problems.

Before solving N+1 problem on post#ndex

Screenshot 2023-08-07 at 17 09 23

Screenshot 2023-08-07 at 17 09 57

 After fixing  N+1 problem on post#ndex

Screenshot 2023-08-07 at 17 11 58

Before solving  N+1 problem on post#show

Screenshot 2023-08-08 at 09 41 28

After fixing  N+1 problem on post#show

Screenshot 2023-08-08 at 09 43 06

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @abiola-arobieke,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* Kindly collaborate with your coding partner(s).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/views/users/show.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  scenario 'show redirects to the post detail page' do
+    click_link Post.last.title
+    expect(page).to have_current_path(user_post_path(@first_user.id, @fourth_post.id))
+  end
+
+  scenario 'redirects to the posts index page' do
+    click_link('See all posts')
+    expect(page).to have_current_path(user_posts_path(@first_user.id))
+  end
+end
```
Comment (DammyShittu) [inline spec/views/users/show.html.erb_spec.rb]: - [ ] The test for checking the number of posts a user has is missing. Kindly add it.

File: spec/views/posts/index.html.erb_spec.rb [hunk 2]
```diff
@@ -1,5 +1,84 @@
...
+
+  scenario 'should show the number of likes for a post' do
+    expect(page).to have_content('Likes: 4')
+    expect(page).to have_content('Likes: 0', count: 3)
+  end
+
+  scenario 'should redirects to the post detail page when a post is clicked' do
+    click_link @second_post.title
+    expect(page).to have_current_path(user_post_path(@first_user, @second_post))
+  end
```
Comment (DammyShittu) [inline spec/views/posts/index.html.erb_spec.rb:83]: - [x] Out of your 27 test cases, this test failed. Kindly fix it.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @abiola-arobieke and @sameedmohsin28,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/show.html.erb_spec.rb (hunk 1 above)
Comment (DammyShittu) [inline spec/views/users/show.html.erb_spec.rb]: - [ ] You marked [this request](https://github.com/abiola-arobieke/blog-app/pull/8#discussion_r1289884136) as resolved but you have not fixed it.

  Kindly add the test for checking the number of posts a user has written in the user show page.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @abiola-arobieke and @sameedmohsin28,

## Highlights
✔️ Passing tests

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @abiola-arobieke, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**